contracts/ERC20WithPermit.sol

```diff
@@ -42,32 +42,21 @@ contract ERC20WithPermit is Ownable, IERC20WithPermit {
     mapping(address => uint256) public override balanceOf;
     mapping(address => mapping(address => uint256)) public override allowance;
 
-    bytes32 public override DOMAIN_SEPARATOR;
+    uint256 public cachedChainId;
```

Comment:
minor, but can we move this below `totalSupply` for gas efficiency?

Reply:
Yes, let's group them together. Please see 012346fdafebca91cc4504912c637d4f0daf701e.